src/focus-zone.ts

```diff
@@ -527,19 +527,35 @@ export function focusZone(container: HTMLElement, settings?: FocusZoneSettings):
           endFocusManagement(...iterateFocusableElements(removedNode, iterateFocusableElementsOptions))
         }
       }
+      // If an element is hidden or disabled, remove it from the list of focusable elements
+      if (mutation.type === 'attributes' && mutation.oldValue === null) {
+        if (mutation.target instanceof HTMLElement) {
+          endFocusManagement(mutation.target)
+        }
+      }
     }
     for (const mutation of mutations) {
       for (const addedNode of mutation.addedNodes) {
         if (addedNode instanceof HTMLElement) {
           beginFocusManagement(...iterateFocusableElements(addedNode, iterateFocusableElementsOptions))
         }
       }
+
+      // Similarly, if an element is unhidden or "enabled", add it to the list of focusable elements
+      // If `mutation.oldValue` is not null, then we may assume that the element was previously hidden or disabled
+      if (mutation.type === 'attributes' && mutation.oldValue !== null) {
+        if (mutation.target instanceof HTMLElement) {
+          beginFocusManagement(mutation.target)
+        }
+      }
     }
   })
 
   observer.observe(container, {
     subtree: true,
     childList: true,
+    attributeFilter: ['hidden', 'disabled'],
+    attributeOldValue: true,
```

Comment:
Will this narrow down the observed set of elements in the first place? Seems like the filter was broader and the `attributeFilter` narrows it down. Just trying to understand how the observer works.

Reply:
`attributeFilter` allows us to watch for any changes of the attributes in the array, in addition to the rules that we already have. So it won't narrow our existing ruleset down, as it will still only run the callback if either a node is added/removed (`childList`), or if an element receives/removes the `hidden`, or `disabled` attributes within the container. This essentially just slightly widens what we watch for.